src/components/Visit/VisitForm/formfields.tsx

```diff
@@ -22,7 +22,9 @@ const FormFields = () => {
 
   const fillValues = (data: OnChangeValue<SelectOption<ClientInfo>, false>) => {
     const clientName = data?.label
-    const latestVisit = visits?.find((v) => v.clientName == clientName)
+    const latestVisit = visits?.pages
+      .flatMap((page) => page)
+      .find((v) => v?.clientName == clientName)
```

Comment:
I might be dumb but shouldnt this value be stored in a state and use `useEffect` to listen to the `clientName`? `useEffect` wont rerun the code on rerender as well.

Reply:
If the value of latestVisit needs to be accessed by multiple components or needs to be used in a way that triggers re-renders, it might be useful to store it in state. In that case, probably could use useState to store the value and useEffect to update it whenever clientName changes.

Reply:
The function that this code appears in is connected to the onChange event for the select field, so it re-runs if `clientName` changes, no need for any additional hooks (and it's unrelated to this PR anyway).

Reply:
Also `latestVisit` isn't used anywhere else, so no need to declare additional state.